vmdb/app/controllers/ops_controller/settings/schedules.rb

```diff
@@ -278,6 +278,85 @@ def build_saved_audit_hash(old_schedule_attributes, new_schedule, add)
     }
   end
 
+  def build_search_filter_from_params
+    case params[:action]
+    when "storage"
+      case params[:filter_typ]
+      when "host"    then {"CONTAINS"    => {"field" => "Storage.hosts-name", "value" => params[:filter_value]}}
+      when "ems"     then {"CONTAINS"    => {"field" => "Storage.ext_management_systems-name", "value" => params[:filter_value]}}
```

Comment:
you changed the order everywhere to be ems, host, storage.
odd that you would change here from that order to a different one